MODULES/non_max_suppression.py

```diff
@@ -109,32 +106,26 @@ def compute_mask_and_index(prob: torch.Tensor,
 
         # this is O(N^2) algorithm
         overlap_measure = NonMaxSuppression.compute_box_intersection_over_min_area(bounding_box=bounding_box)
-        binarized_overlap_measure = (overlap_measure > overlap_threshold).float()
-        assert binarized_overlap_measure.shape == (n_boxes, n_boxes, batch_size)
-
-        # The noise need to be added to the probabilities
-        noisy_score = (1-randomize_nms_factor) * prob + randomize_nms_factor * torch.rand_like(prob)
-        # noisy_score = torch.clamp(prob + randomize_nms_factor * torch.randn_like(prob), min=0.0)
+        binarized_overlap = (overlap_measure > overlap_threshold).float()
+        assert binarized_overlap.shape == (n_boxes, n_boxes, batch_size)
 
         if topk_only:
             # If nms_mask = 1 then this is equivalent to do topk only
-            chosen_nms_mask = torch.ones_like(noisy_score)
+            chosen_nms_mask = torch.ones_like(prob)
         else:
-            possible = (noisy_score > score_threshold).float()
-            chosen_nms_mask: torch.Tensor = NonMaxSuppression.perform_nms_selection(mask_overlap=binarized_overlap_measure,
-                                                                                    score=noisy_score,
+            possible = (prob > 0.0).float()
```

Comment:
At the beginning all prob should be possible so that they have a chance of receiving gradients.
Change (prob > 0.0).float() -> (prob > -1.0).float()